src/engine/strat_engine/liminal/liminal.rs

```diff
@@ -479,6 +481,63 @@ impl LiminalDevices {
         }
     }
 
+    /// On udev events, stratisd checks whether block devices have changed in size.
+    /// This allows us to update the user with the new size if it has changed.
+    /// This method runs a check on device size on each udev change or add event
+    /// to determine whether the size has indeed changed so we can update it in
+    /// our internal data structures.
+    pub fn block_evaluate_size(
+        pools: &mut Table<PoolUuid, StratPool>,
+        event: &UdevEngineEvent,
+    ) -> StratisResult<Option<(DevUuid, StratBlockDevDiff)>> {
+        let mut ret = None;
+
+        let event_type = event.event_type();
+        let device_path = match event.device().devnode() {
+            Some(d) => d,
+            None => return Ok(None),
+        };
+        let device_info = match event_type {
+            libudev::EventType::Add | libudev::EventType::Change => {
+                if device_path.exists() {
+                    identify_block_device(event)
+                } else {
+                    None
+                }
+            }
+            _ => None,
+        };
+
+        if event_type == libudev::EventType::Add || event_type == libudev::EventType::Change {
+            if let Some(di) = device_info {
+                let pool_uuid = di.stratis_identifiers().pool_uuid;
+                let dev_uuid = di.stratis_identifiers().device_uuid;
+                if let Some((_, pool)) = pools.get_mut_by_uuid(pool_uuid) {
+                    if let Some((tier, dev)) = pool.get_mut_strat_blockdev(dev_uuid)? {
```

Comment:
Is it still correct for the ```get_mut_strat_blockdev``` function definition to be decorated with ``` #[pool_mutating_action("NoRequests")]```? It is called only in this one context.

Reply:
I have to think about this a little bit. Currently, `NoRequests` is really used for preventing IPC actions and this isn't an IPC action, so it may be better to change it to `NoPoolChanges` instead given that this is now a background task.

Reply:
Does that sound like the right change to you?

Reply:
Yes. I think that would work out fine.